herbrain/pregnancy/gpt_chat.py

```diff
@@ -0,0 +1,184 @@
+                html.P(
+                    "Ask questions about the brain changes during pregnancy, hormone levels, or any other aspects of the data shown on this page. The AI neurobot will analyze the 3D visualization of your brain and provide insights.",
+                    style={"fontSize": "0.9em", "fontFamily": S.text_fontfamily, "color": "#666"}
+                ),
+                # Input area
+                dbc.Row([
+                    dbc.Col([
+                        dbc.Textarea(
+                            id="gpt-input",
+                            placeholder="Type your question here...",
+                            style={"width": "100%", "height": "100px", "marginBottom": "10px"}
+                        ),
+                        dbc.Button("Ask ChatGPT", id="gpt-submit", color="primary", className="mb-3"),
+                    ])
+                ]),
+                # Chat history container
+                html.Div(
+                    id="chat-history",
+                    style={
+                        "height": "200px",  # Reduced from 400px to 200px
+                        "overflowY": "auto",
+                        "border": "1px solid #ddd",
+                        "borderRadius": "5px",
+                        "padding": "10px",
+                        "marginBottom": "10px",
+                        "backgroundColor": "#f8f9fa"
+                    }
+                ),
+                # Store for chat history
+                dcc.Store(id="chat-store", data=[]),
+            ])
+        ])
+    ])
+
+def create_message_bubble(message, is_user=True):
+    """Create a message bubble for the chat."""
+    return html.Div(
+        [
+            html.Div(
+                message,
+                style={
+                    "backgroundColor": "#007bff" if is_user else "#e9ecef",
+                    "color": "white" if is_user else "black",
+                    "padding": "10px",
+                    "borderRadius": "10px",
+                    "marginBottom": "10px",
+                    "maxWidth": "80%",
+                    "marginLeft": "auto" if is_user else "0",
+                    "marginRight": "0" if is_user else "auto",
+                }
+            )
+        ],
+        style={"marginBottom": "10px"}
+    )
+
+@callback(
+    [Output("chat-history", "children"),
+     Output("gpt-input", "value")],
+    Input("gpt-submit", "n_clicks"),
+    [State("gpt-input", "value"),
+     State("chat-store", "data"),
+     State("gestWeek-slider", "value"),  # Gestational week slider
+     State("estro-slider", "value"),     # Estrogen slider
+     State("prog-slider", "value"),      # Progesterone slider
+     State("lh-slider", "value"),        # LH slider
+     State("mesh-plot", "figure")],      # Current mesh figure
+    prevent_initial_call=True
+)
+def update_chat(n_clicks, question, chat_history, gest_week, estro, prog, lh, figure):
```

Comment:
how responsive is this? I'm afraid not fast enough for real-time

should it be updated every time or should it have its own inputs?